pallets/storage-proofs/src/lib.rs

```diff
@@ -387,67 +390,20 @@ impl InherentError {
     }
 }
 
-// TODO: Move this to Storage Providers pallet.
-/// A trait to lookup registered Storage Providers.
+/// A trait to lookup registered Providers, their Merkle Patricia Trie roots and their stake.
 ///
-/// It is abstracted over the `AccountId` type, `StorageProvider` type, total number of users
-/// and Balance type.
-pub trait StorageProvidersInterface {
+/// It is abstracted over the `AccountId` type, `Provider` type, `Balance` type and `MerkleHash` type.
+pub trait ProvidersInterface {
     /// The type which can be used to identify accounts.
     type AccountId: Parameter + Member + MaybeSerializeDeserialize + Debug + Ord + MaxEncodedLen;
-    /// The type corresponding to the staking balance of a registered Storage Provider.
+    /// The type which represents a registered Provider.
+    type Provider: Parameter + Member + MaybeSerializeDeserialize + Debug + Ord + MaxEncodedLen;
+    /// The type corresponding to the staking balance of a registered Provider.
     type Balance: fungible::Inspect<Self::AccountId>
         + fungible::hold::Inspect<Self::AccountId>
         + fungible::freeze::Inspect<Self::AccountId>;
```

Comment:
I get this Balance type but at the same time I'm not sure I can return a value that has all three traits in itself, which is why I think there's an issue with the return type below here